### PR TITLE
Api Key in adt/code/test.py, Log Works as well

### PR DESCRIPTION
**Pull request by MarcPfeif**

### CODE REVIEW

**Comment by waltertschwe** [comment]
i removed all the .pyc files and "__pycache__" directories" on the master branch.  I updated .gitignore so we don't include these files.  We don't want these to be part of the code base.  Merge the master branch into your branch to get the updates.

File: app/adt/mssqlserver.py [hunk 1]
```diff
@@ -16,7 +17,12 @@ def __init__(self):
         self.driver = os.getenv("MSSQL_DRIVER")
         self.username = os.getenv("MSSQL_USERNAME")
         self.password = os.getenv("MSSQL_PASSWORD")
-    
+
```
Comment (waltertschwe) [inline app/adt/mssqlserver.py]: these values need to be pulled from the .env file

File: app/adt/views.py [hunk 2]
```diff
@@ -2,46 +2,41 @@
...
+        else:
+            # No backend authenticated the credentials
+            return JsonResponse(["ERROR","Invalid API KEY"], safe =False)
 
 # uses django Authenticate to authenticate username and password
 def Auth(request):
-    user = authenticate(username=request.GET['username'], password=request.GET['password'])
-    if user is not None:
+
+    print(request.GET['APIKEY'])
```
Comment (waltertschwe) [inline app/adt/views.py]: remove this print.  if necessary use logging.info($stuffToLog)

File: app/app/settings.py [hunk 3]
```diff
@@ -83,7 +83,9 @@
     'default': {
         'ENGINE': 'django.db.backends.postgresql',
         'NAME': 'ADT',
-    
+        'USER':'postgres',
```
Comment (waltertschwe) [inline app/app/settings.py]: this needs to be removed

File: env [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+## DW STAGING
```
Comment (waltertschwe) [inline env]: this file needs to be removed

File: env-file [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
+## DW STAGING
+MSSQL_SERVER=192.168.254.239
```
Comment (waltertschwe) [inline env-file]: remove this file too

**Comment by MarcPfeif** [comment]
I've fixed the requested changes.

File: app/adt/views.py (hunk 2 above)
Comment (waltertschwe) [reply app/adt/views.py]: this is fine because it's a test file

**Comment by waltertschwe** [comment]
everything looks good.  you just need to rebase the branch with master to get rid of all these conflicts.

``` git checkout $feature branch
git merge master 
```

**Comment by MarcPfeif** [comment]
Ok Just did ,Check it over once more

File: app/adt/adt.py [hunk 6]
```diff
@@ -29,6 +29,8 @@ def get_discharge_adt(self, time):
 
         return results
 
+<<<<<<< HEAD
```
Comment (waltertschwe) [inline app/adt/adt.py]: have to resolve the conflicts.

File: app/adt/adt.py [hunk 7]
```diff
@@ -41,6 +43,7 @@ def get_admit_adt(self, time):
 
         return results
 
+>>>>>>> master
```
Comment (waltertschwe) [inline app/adt/adt.py]: have to resolve the conflicts.

File: app/adt/urls.py [hunk 8]
```diff
@@ -2,9 +2,15 @@
 from . import views
 
 urlpatterns = [
+<<<<<<< HEAD
```
Comment (waltertschwe) [inline app/adt/urls.py]: have to resolve the conflicts.

File: app/adt/urls.py [hunk 9]
```diff
@@ -2,9 +2,15 @@
 from . import views
 
 urlpatterns = [
+<<<<<<< HEAD
+    # path('discharge/', views.discharge_all, name = 'discharge_all' ),
+    # path('admit/',views.admit_all, name = 'admit_all'),
+    path('status/', views.get_all, name = 'get_all'),
+=======
```
Comment (waltertschwe) [inline app/adt/urls.py]: have to resolve the conflicts.

File: app/adt/urls.py [hunk 10]
```diff
@@ -2,9 +2,15 @@
...
+    # path('discharge/', views.discharge_all, name = 'discharge_all' ),
+    # path('admit/',views.admit_all, name = 'admit_all'),
+    path('status/', views.get_all, name = 'get_all'),
+=======
     path('discharge/', views.discharge_all, name = 'discharge_all' ),
     path('admit/',views.admit_all, name = 'admit_all'),
     path('test-data/admit-discharge',views.test_admit_discharge, name='test_admit_discharge'),
 
 
+>>>>>>> master
```
Comment (waltertschwe) [inline app/adt/urls.py]: have to resolve the conflicts.

File: app/adt/views.py [hunk 11]
```diff
@@ -6,13 +6,21 @@
...
 from django.contrib.auth.models import User, Permission
+from .models import Log, ApiKey
 from django.contrib.auth import authenticate
 from os.path import join, dirname, os
 
 
 from .adt import Adt
 
 #return all discharges for today
+<<<<<<< HEAD
```
Comment (waltertschwe) [inline app/adt/views.py]: have to resolve the conflicts.

File: app/adt/views.py [hunk 12]
```diff
@@ -6,13 +6,21 @@
...
+def get_all(request):
+
+        if Auth(request):
+            #backend authenticated the credentials
+            time = datetime.fromtimestamp(int(request.GET['time'])/1000.0).strftime('%Y-%m-%d %H:%M:%S.%f')
+=======
```
Comment (waltertschwe) [inline app/adt/views.py]: have to resolve the conflicts.

File: app/adt/views.py [hunk 13]
```diff
@@ -43,17 +51,36 @@ def admit_all(request):
     else:
         # No backend authenticated the credentials
         return JsonResponse(["ERROR","Invalid Username/password"], safe =False)
+>>>>>>> master
```
Comment (waltertschwe) [inline app/adt/views.py]: have to resolve the conflicts.

File: app/adt/views.py [hunk 14]
```diff
@@ -43,17 +51,36 @@ def admit_all(request):
...
+    if ApiKey.objects.filter(key = request.GET['APIKEY']).exists():
+        log(request,True)
         return True
-    else:
 
+    else:
+        log(request, False)
         return False
 
+<<<<<<< HEAD
```
Comment (waltertschwe) [inline app/adt/views.py]: have to resolve the conflicts.

File: app/adt/views.py [hunk 15]
```diff
@@ -43,17 +51,36 @@ def admit_all(request):
...
+
+def log(request, status):
+
+    log = Log(api_key = request.GET['APIKEY'], date = datetime.now(), ip = request.environ['REMOTE_ADDR'], approved = status)
+    log.save()
+=======
```
Comment (waltertschwe) [inline app/adt/views.py]: have to resolve the conflicts.

File: app/adt/views.py [hunk 16]
```diff
@@ -65,3 +92,4 @@ def test_admit_discharge(request):
     response['Content-Disposition'] = "filename=%s" % data
 
     return response
+>>>>>>> master
```
Comment (waltertschwe) [inline app/adt/views.py]: have to resolve the conflicts.

**Comment by waltertschwe** [comment]
LGTM, there are a few .pyc files still in but I will remove when we merge to master.

**Pull request merged**